UnityGLTF/Assets/UnityGLTF/Runtime/Scripts/SceneImporter/ImporterMeshes.cs

```diff
@@ -34,9 +48,83 @@ protected virtual async Task ConstructMesh(GLTFMesh mesh, int meshIndex, Cancell
 				return;
 			}
 
+			var firstPrim = mesh.Primitives.Count > 0 ?  mesh.Primitives[0] : null;
+
+#if UNITY_DRACO
+			var anyHadDraco = false;
+			Mesh[] meshes = new Mesh[mesh.Primitives.Count];
+			var meshDataArray = Mesh.AllocateWritableMeshData(mesh.Primitives.Count);
+			var dracoDecodeResults = new DracoMeshLoader.DecodeResult[mesh.Primitives.Count];
+			for (int i = 0; i < mesh.Primitives.Count; i++)
+			{
+				var primitive = mesh.Primitives[i];
+				if (primitive.Extensions == null || !primitive.Extensions.ContainsKey("KHR_draco_mesh_compression"))
+					continue;
+
+				anyHadDraco = true;
+
+				if (primitive.Extensions.TryGetValue("KHR_draco_mesh_compression", out var extension))
+				{
+
+					var dracoExtension = (KHR_draco_mesh_compression) extension;
+					if (_assetCache.BufferCache[dracoExtension.bufferView.Value.Buffer.Id] == null)
+						await ConstructBuffer(dracoExtension.bufferView.Value.Buffer.Value, dracoExtension.bufferView.Value.Buffer.Id);
+
+					BufferCacheData bufferContents = _assetCache.BufferCache[dracoExtension.bufferView.Value.Buffer.Id];
+
+
+					GLTFHelpers.LoadBufferView(dracoExtension.bufferView.Value, bufferContents.ChunkOffset,
+						bufferContents.Stream, out byte[] bufferViewData);
+
+					int weightsAttributeId = -1;
+					if (dracoExtension.attributes.ContainsKey("WEIGHTS_0"))
+						weightsAttributeId = (int) dracoExtension.attributes["WEIGHTS_0"];
+
+					int jointsAttributeId = -1;
+					if (dracoExtension.attributes.ContainsKey("JOINTS_0"))
+						jointsAttributeId = (int) dracoExtension.attributes["JOINTS_0"];
+
+					// TODO: check if normals and tangents are needed
```

Comment:
Can be a follow-up PR but could you please add more info on what the TODO here encompasses?